### PR TITLE
Data structures algorithms

### PR DESCRIPTION
**Pull request by peter-bachowski**
## Because

Change the position of four links in the Assignment section, as well as one link in the Additional Resources section. Also, remove one link from the Additional Resources section.

## This PR

- Change link placement from [this video] to [Coursera’s Algorithms course].
- Change link placement from [this video] to [how binary search works].
- Change link placement from [this video] to [how a binary search tree is constructed from an unordered array].
- Change link placement from [this video] to [principles of queues and stacks].
- Delete line: [An article] that discusses how to construct a binary search tree from an unordered array.
- Change link placement from [A stack overflow discussion] to [relative strengths of BFS and DFS].

## Issue

Related to #28290

## Pull Request Requirements

-   [x] I have thoroughly read and understand [The Odin Project curriculum contributing guide](https://github.com/TheOdinProject/curriculum/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`
-   [x] The `Because` section summarizes the reason for this PR
-   [x] The `This PR` section has a bullet point list describing the changes in this PR
-   [x] If this PR addresses an open issue, it is linked in the `Issue` section
-   [x] If any lesson files are included in this PR, they have been previewed with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure it is formatted correctly
-   [x] If any lesson files are included in this PR, they follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

### CODE REVIEW

File: ruby/computer_science/common_data_structures_algorithms.md [hunk 1]
```diff
@@ -32,11 +32,12 @@ This section contains a general overview of topics that you will learn in this l
 
 <div class="lesson-content__panel" markdown="1">
   1. Glance over the [Wikipedia entry on Data Structures](http://en.wikipedia.org/wiki/Data_structure) for a high level overview of things.
-  2. Learn about basic algorithms from Coursera's Algorithms course in [this video](http://www.youtube.com/watch?v=u2TwK3fED8A).  The first 10 minutes are really the meat of the introduction to algorithms, the rest gets more mathematical (if you're so inclined).
+  2. Learn about basic algorithms from [Coursera’s Algorithms course](http://www.youtube.com/watch?v=u2TwK3fED8A).
+  The first 10 minutes are really the meat of the introduction to algorithms, the rest gets more mathematical (if you're so inclined).
```
Comment (MaoShizhong) [inline ruby/computer_science/common_data_structures_algorithms.md:36]: No need for the line break here
```suggestion
  2. Learn about basic algorithms from [Coursera’s Algorithms course](http://www.youtube.com/watch?v=u2TwK3fED8A). The first 10 minutes are really the meat of the introduction to algorithms, the rest gets more mathematical (if you're so inclined).
```

**Review by MaoShizhong — CHANGES_REQUESTED** [review]
Many thanks @peter-bachowski
There's just the one required change detailed in the comment below.

**Optional:** There are some linting errors in this lesson file (not introduced by you). While we're at it, would you be happy to fix them after you've made the required change? If so, let me know when you've pushed those fixes, otherwise let me know if not so I know not to expect them.

If you were happy to fix the linting errors, we have [instructions for linting and fixing locally](https://github.com/TheOdinProject/curriculum/blob/main/CONTRIBUTING.md#curriculum-linting) you can follow. I think in this instance, you'd need to run the `fix:lesson` script twice to catch and fix all the errors.

File: ruby/computer_science/common_data_structures_algorithms.md (hunk 1 above)
Comment (peter-bachowski) [reply ruby/computer_science/common_data_structures_algorithms.md]: Hey @MaoShizhong thanks for taking the time to review my changes! The code should be clean now after I used the linter, Check it out!

Comment (MaoShizhong) [reply ruby/computer_science/common_data_structures_algorithms.md]: Thanks - the line break described above is still present and should be resolved, then we should be good to merge after.

Comment (peter-bachowski) [reply ruby/computer_science/common_data_structures_algorithms.md]: Should be good now Mao, let me know if there's anything else. Thanks for being patient!

**Review by MaoShizhong — APPROVED** [review]
:rocket:

**Pull request merged**